nemo/collections/common/data/lhotse/dataloader.py

```diff
@@ -449,6 +478,58 @@ def measure_length(self, example: Any) -> float:
         raise RuntimeError(f"Unsupported example type: {type(example)}")
 
 
+@dataclass
+class FixedBucketBatchSizeConstraint2D(FixedBucketBatchSizeConstraint):
```

Comment:
I'm just going to trust these and skip over reading them in depth.

Reply:
This turned out to be where the bugs were 😅